### DISCUSSION TITLE
Problema com questao inicial de curso

### DISCUSSION THREAD

**Discussion by Pecoits1** [frontendbr/forum#2082] [Perguntas e Respostas]
Bom dia pessoal...estou começando um curso de JS e me deparei com essa questão e eu não consegui resolver de maneira alguma...alguém poderia me ajudar?

Neste desafio, você deve implementar a função calculadoraAdicaoSubtracao, que:
Recebe 3 argumentos sendo os dois primeiros números inteiros, numero e outro numero e o terceiro operacao, sendo uma string "+" ou "-"

Retorna um valor dependendo da operacao:
- Se a operacao é "+", então retorna a soma do numero com o outroNumero.
- Se a operacao é "-", entao retorna a subtração do numero com outroNumero.
- Se operacao não é nenhuma das operações esperadas, então retorna uma sting "operação invalida"

**Comment by igorlvicente** [comment]
Até onde chegaste?

**Comment by Pecoits1** [comment]
Bom dia Igor,

Com um colega imaginamos algo assim:

function calculadoraAdicaoSubtracao(num1, num2, soma = '+', subtracao =
'-'){

soma = prompt("Informe uma operação: ")
subtracao = prompt("Informe outra operação: ")

if((soma == "+") && (subtracao == '-')){
    return num1 + num2
    return num1 - num2
}
else{
   console.log("Operacão Inválida!")
}
}

Porém no teste simplesmente não funciona...

Obrigado
Em qui, 13 de mai de 2021 07:05, Igor L. Vicente ***@***.***>
escreveu:

**Reply by igorlvicente** [reply]
Colocaste a pergunta dentro da função e não estás chamando-a. Esse é o problema principal.
Há alguns outros mas esses eu acredito que descobrirás sozinho.

**Comment by sou-gabriel** [comment]
>   if ((soma == "+") && (subtracao == '-')) {
    return num1 + num2
    return num1 - num2
  }

Amigo o return é utilizado em 2 situações:
  1. Parar a execução de uma função;
  2. Fazer a função retornar um valor.

Esse seu pedaço de código é inválido, o segundo return nunca será acionado já que há um return acima dele. Esse é um ponto. Uma solução para este caso seria um ternário, mas pode haver outros, caberá a sua curiosidade em se aprofundar nessa questão.

**Reply by sou-gabriel** [reply]
> soma = prompt("Informe uma operação: ")
  subtracao = prompt("Informe outra operação: ")

Outra observação... Você está fazendo um mal uso dos parâmetros, eles são geralmente usados para receberem valores que estão vindo de fora da função. Uma dica que te dou neste caso é remover essas duas linhas e passar o operador e os operandos como argumentos da invocação da função e dentro dela você apenas verifica com por meio de estruturas if/switch, por exemplo, qual valor aquele(es) parâmetros armazena(am). Espero ter ajudado!
